src/sql/services/notebook/notebookServiceImpl.ts

```diff
@@ -18,42 +18,138 @@ import { RenderMimeRegistry } from 'sql/parts/notebook/outputs/registry';
 import { standardRendererFactories } from 'sql/parts/notebook/outputs/factories';
 import { LocalContentManager } from 'sql/services/notebook/localContentManager';
 import { SessionManager } from 'sql/services/notebook/sessionManager';
-import { Extensions, INotebookProviderRegistry } from 'sql/services/notebook/notebookRegistry';
+import { Extensions, INotebookProviderRegistry, NotebookProviderDescription } from 'sql/services/notebook/notebookRegistry';
 import { Emitter, Event } from 'vs/base/common/event';
+import { Memento } from 'vs/workbench/common/memento';
+import { IStorageService } from 'vs/platform/storage/common/storage';
+import { IExtensionService } from 'vs/workbench/services/extensions/common/extensions';
+import { IExtensionManagementService, IExtensionIdentifier } from 'vs/platform/extensionManagement/common/extensionManagement';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { getIdFromLocalExtensionId } from 'vs/platform/extensionManagement/common/extensionManagementUtil';
+import { Deferred } from 'sql/base/common/promise';
 
+export interface NotebookProviderProperties {
+	provider: string;
+	fileExtensions: string[];
+}
+
+interface NotebookProviderCache {
+	[id: string]: NotebookProviderProperties;
+}
+
+interface NotebookProvidersMemento {
```

Comment:
This memento is based on capabilities service. I'm using this since it should be a short hop over to fully caching the results instead of having to wait on provider registration in the future